File: server/src/main/java/com/example/server/note/domain/Date.java
package com.example.server.note.domain;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.Column;
import java.time.LocalDateTime;

@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public abstract class Date {
    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;

    // getter ...
}

File: server/src/main/kotlin/work/notespace/server/note/entity/Date.kt
package work.notespace.server.note.entity

import jakarta.persistence.Column
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
abstract class Date {
    @CreatedDate
    @Column(name = "CREATED_AT", updatable = false, nullable = false)
    val createdAt: LocalDateTime? = null

    @LastModifiedDate
    @Column(name = "UPDATED_AT", nullable = false)
    var updatedAt: LocalDateTime? = null
}